src/misc/bleeding.ts

```diff
@@ -0,0 +1,197 @@
+import { ParticleBuilder } from 'com.destroystokyo.paper';
+import { Location, Material, Particle } from 'org.bukkit';
+import { BlockFace } from 'org.bukkit.block';
+import { Waterlogged } from 'org.bukkit.block.data';
+import { Entity, EntityType, LivingEntity, Player } from 'org.bukkit.entity';
+import { EntityDamageEvent } from 'org.bukkit.event.entity';
+import { DamageCause } from 'org.bukkit.event.entity.EntityDamageEvent';
+
+/**
+ * Add a bleeding effect for damageable entities by randomly generating blood puddles and splashes around entity
+ *
+ * Currently implemented features:
+ *
+ *  - Damage amount is relative to how much and how long entity will bleed
+ *  - Bleeding duration and intensity are increased when entity is damaged again while already bleeding
+ *  - Bleeding will exponentially decrease
+ *  - Armor, clothing etc... will reduce bleeding amount
+ *
+ * @author Juffel
+ */
+
+const TIMER_DELAY = 1000; // timer delay in millis
+const BLOOD_MATERIAL = Material.DEAD_BUBBLE_CORAL_FAN; // blood material
+const PARTICLE_DATA = Material.REDSTONE_BLOCK.createBlockData(); // block data for the blood particle
+const PARTICLE_AMOUNT = 5; // amount of particles
+
+// entities that don't bleed
+const ENTITY_BLACKLIST = [
```

Comment:
Prefer ES6 Set
```suggestion
const ENTITY_BLACKLIST = new Set([ 
    EntityType.CREEPER, 
    EntityType.SKELETON, 
    ... 
    ])
```

Then replace LIST.includes(...)   with    LIST.has(...)